dcos/config.py

```diff
@@ -52,24 +54,36 @@ def get_config(mutable=False):
     return load_from_path(path, mutable)
 
 
-def get_config_vals(keys, config=None):
-    """Gets config values for each of the keys.  Raises a DCOSException if
-    any of the keys don't exist.
+def get_config_val(name, config=None):
+    """Returns the config value for the specified key. Looks for corresponding
+    environment variable first, and if it doesn't exist, uses the config value.
+    - "core" properties get resolved to env variable DCOS_SUBKEY. With the
+    exception of subkeys that already start with DCOS, in which case we look
+    for SUBKEY first, and "DCOS_SUBKEY" second, and finally the config value.
+    - everything else gets resolved to DCOS_SECTION_SUBKEY
 
+    :param name: name of paramater
+    :type name: str
     :param config: config
     :type config: Toml
-    :param keys: keys in the config dict
-    :type keys: [str]
-    :returns: values for each of the keys
-    :rtype: [object]
+    :returns: value of 'name' parameter
+    :rtype: object | None
     """
 
-    config = config or get_config()
-    missing = [key for key in keys if key not in config]
-    if missing:
-        raise missing_config_exception(keys)
+    if config is None:
+        config = get_config()
+
+    section, subkey = split_key(name.upper())
+
+    env_var = None
+    if section == "CORE":
+        env_var = "DCOS_{}".format(subkey)
+        if subkey.startswith("DCOS") and os.environ.get(subkey):
+            env_var = subkey
```

Comment:
This is a bit confusing to read. Maybe we can write this as:

```
if subkey.startswith("DCOS") and os.environ.get(subkey):
    env_var = subkey
else:
    env_var = "DCOS_{}".format(subkey)
```

Let's write a short comment as to why we are doing this.